src/helics/apps/Recorder.cpp

```diff
@@ -24,12 +23,22 @@ SPDX-License-Identifier: BSD-3-Clause
 #include <map>
 #include <regex>
 #include <set>
+#include <spdlog/spdlog.h>
 #include <stdexcept>
 #include <string>
 #include <thread>
 #include <utility>
 #include <vector>
 
+/** encode the string in base64 if needed otherwise just return the string*/
+static std::string encode(const std::string& str2encode)
+{
+    return std::string("b64[") +
+        gmlc::utilities::base64_encode(reinterpret_cast<const unsigned char*>(str2encode.c_str()),
+                                       static_cast<int>(str2encode.size())) +
+        ']';
+}
+
```

Comment:
This function seems like it should eventually get moved to `gmlc/utilities/base64.h`

Reply:
probably, though I am not sure b64[ is fully standard,  I should look into that more.

Reply:
There isn't really a standard for marking base64 data in JSON, other than just using the base64 encoded data and knowing that the field is base64 encoded.. which it looks like we have an encoding field already that gets set `message["encoding"] = "base64";` that could get used.